### ISSUE TITLE
[SUGGESTION] Battle Mode Hotkey

### ISSUE THREAD

**Issue by Senakysam** [tsunamayo/Skywanderers#1343]
In game, we have a hotkey for landing so that the AI knows how to use a player-created ship and work correctly. I think the same should be done with a battle mode. Some players have logic on their ship that needs to be activated in order to fight. Turrets pop out, missile doors open, and blast doors close. In order for the AI to use these ships effectively, we need a hotkey that can be set up so that when the AI thinks it is in battle, it will activate the logic to deploy everything.

**Comment by ProPeach** [comment]
This sounds like a great catch all feature to help out AI, and fits well with the current system too!

**Comment by Dwarf-LordPangolin** [comment]
This sounds like a really smart idea. It could even be linked to the shield generators too, so they aren't on and making waste heat when they're not needed.

**Comment by tsunamayo** [comment]
Okay I am not sure 100% on all the details, I need to think this through a bit more. But I see the potential definitely. Question is does it would require a dedicated key like the landing gear? (seems a bit overkill). 
I could have such a safe lock trigger when you are close a space station also. If that is the case I could also merge this with #1339, where this query key could be triggered by various internal state, AI - Combat mode being one.

**Comment by Senakysam** [comment]
I wasn't really considering weapon safety locks with this suggestion, though I guess that can be a thing. It was more about how a lot of builders will have turrets retract and missile doors close when not in combat, and then pop out and open when entering combat. This is easy for a player to do since they can read the ship manual or the button labels, but an AI can't do it. So the suggestion of a Hotkey is so that the AI has a singular switch to trigger and it knows what it does, that the builder can hook up all the necessary logic mechanisms to. Otherwise, if an AI tries to use these ships in battle, they end up severely hampered from what the builder intended since the turrets and weapons are unable to turn or shoot.

This is just like hooking up landing gear to the Landing Gear Hotkey. Without it, the AI wouldn't know what switches are for landing and what are for anything else.

**Comment by tsunamayo** [comment]
Yes I totally get the use Sen, I need a bit of time to think about an unified approach. I guess after a while we will realize there is ton of other use case of things that people would like to see their AI do or even just simply see their ship do automatically even when they are at the command
For example to trigger something when a landing event happens: what about opening the main hatch to get the crew out ect?
The more I think about this the closer the Hotkey and Query looks, I was think of maybe merging them, and add a lot more options. Depending on the system you link them to you would have different signal / event to choose from. 
Hotkey1 /2 ect being a special case you get when you link the query / hotkey gate from a yoke. Hope it make sense.

**Comment by Senakysam** [comment]
Ah, okay.
